File: svc-position/src/main/java/com/beancounter/position/accumulation/AccumulationStrategy.java
package com.beancounter.position.accumulation;

import com.beancounter.common.model.Portfolio;
import com.beancounter.common.model.Position;
import com.beancounter.common.model.Trn;

public interface AccumulationStrategy {

  void accumulate(Trn trn, Portfolio portfolio, Position position);
}

File: svc-position/src/main/java/com/beancounter/position/accumulation/AccumulationStrategy.kt
package com.beancounter.position.accumulation

import com.beancounter.common.model.Portfolio
import com.beancounter.common.model.Position
import com.beancounter.common.model.Trn

interface AccumulationStrategy {
    fun accumulate(trn: Trn, portfolio: Portfolio, position: Position)
}
